src/controller/index.ts

```diff
@@ -233,14 +219,25 @@ export default class SimulariumController {
         }
     }
 
-    public convertTrajectory(
+    public convertAndLoadTrajectory(
+        netConnectionConfig: NetConnectionParams,
         obj: Record<string, unknown>,
         fileType: string
     ): Promise<void> {
-        if (!this.converter) {
-            return Promise.reject();
+        try {
+            this.configureNetwork(netConnectionConfig);
+            if (!(this.simulator instanceof RemoteSimulator)) {
+                throw new Error("Autoconversion requires a RemoteSimulator");
+            }
+        } catch (e) {
+            return Promise.reject(e);
         }
-        return this.converter.convertTrajectory(obj, fileType);
+
+        return this.simulator.convertTrajectory(obj, fileType).then(() => {
```

Comment:
TODO: check the status of this potential race condition, at least leave a comment indicating the potential

Reply:
if it's not easy to create the race condition now, then it might make sense to defer it to a new issue/PR that is resolved soon-ish.  one thing we used to talk about a lot is having every message contain the id of the simulation it's dealing with, and so you can do checks on both ends to make sure the id matches what the server or client thinks is current for the given connection.